galsim/config/extra_psf.py

```diff
@@ -17,62 +17,100 @@
 #
 
 import galsim
+import math
+import numpy
+import logging
 
-def DrawPSFStamp(psf, config, bounds, offset, method):
+# The psf extra output type builds an Image of the PSF at the same locations as the galaxies.
+
+# The code the actually draws the PSF on a postage stamp.
+def DrawPSFStamp(psf, config, base, bounds, offset, method, logger=None):
     """
     Draw an image using the given psf profile.
 
     @returns the resulting image.
     """
-
-    if not psf:
-        raise AttributeError("DrawPSFStamp requires psf to be provided.")
-
-    if ('output' in config and 'psf' in config['output'] and 
-        'draw_method' in config['output']['psf'] ):
-        method = galsim.config.ParseValue(config['output']['psf'],'draw_method',config,str)[0]
+    if 'draw_method' in config:
+        method = galsim.config.ParseValue(config,'draw_method',base,str)[0]
         if method not in ['auto', 'fft', 'phot', 'real_space', 'no_pixel', 'sb']:
             raise AttributeError("Invalid draw_method: %s"%method)
     else:
         method = 'auto'
 
-    # Special: if the galaxy was shifted, then also shift the psf 
-    if 'shift' in config['gal']:
-        gal_shift = galsim.config.GetCurrentValue(config['gal'],'shift')
-        if False:
-            logger.debug('obj %d: psf shift (1): %s',config['obj_num'],str(gal_shift))
+    # Special: if the galaxy was shifted, then also shift the psf
```

Comment:
Done.  655428d

I kept demo5 with the same functionality, so it uses the new psf.shift = galaxy option.  

It turns out this change also affected demo10, which has a psf output and a galaxy shift.  So in that case, I took the opportunity to switch to an explicit offset parameter rather than copying the galaxy's shift, which both seems pedagogically useful and covers the missing functionality test.